logger.go

```diff
@@ -27,58 +27,81 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 )
 
 type LoggingClient struct {
 	owningServiceName string
-	RemoteUrl         string
-	LogFilePath       string
+	remoteEnabled     bool
+	logTarget         string
 	stdOutLogger      *log.Logger
 	fileLogger        *log.Logger
 }
 
 // Create a new logging client for the owning service
-func NewClient(owningServiceName string, remoteUrl string) LoggingClient {
+func NewClient(owningServiceName string, isRemote bool, logTarget string) LoggingClient {
```

Comment:
I know it is already closed, but I will make some comments anyway.

Why it needs to be exclusive(local or remote)? If the remoteUrl is not empty, it will log to remote. Also if local filepath is not empty it will also log locally.

Reply:
In my view, explicit intent is better than implicit. In the case of supplying both file and URL paths, if I as a consumer of the logging client want to turn remote logging off and there isn't an explicit flag to do so, how do I know if it can be done? I have to go look at the code for the logging-client library to answer that question. As a consumer of someone else's library, I shouldn't have to do that. The provider of the library should present me with the discrete options that indicate the library's capabilities. And in fact, I would argue that this Configuration struct in core-data is too inclusive. The concerns aren't separated. Core-data shouldn't own the properties on this struct which are pertinent to the logging-client and are then passed in as parameters. Instead, the library should expose a public type that I as a consumer can set properties on in order to configure the library. So I'm actually restraining myself in limiting my changes to what you see above.